drift/info_parser.py

```diff
@@ -21,8 +22,10 @@ def _select_applicable_info(systems, fact_namespace):
     # create dicts of id + info
     ids_and_info = [_system_facts_and_id(system, fact_namespace) for system in systems]
 
-    # TODO: we are assuming all info keys exist for each system. This is not a good assumption
-    available_info_names = ids_and_info[0].keys()
+    # union the keys into one big set
+    available_info_names = set()
+    for fact_ids in ids_and_info:
+        available_info_names |= fact_ids.keys()
```

Comment:
I tried a set comprehension for this, but `keys()` is an iterator and needs to be unioned to convert it to a set. The for loop was more readable.